hw/opentitan/ot_usbdev.c

```diff
@@ -708,12 +1469,117 @@ static void ot_usbdev_write_usbctrl(OtUsbdevState *s, uint32_t val32)
         if (ot_usbdev_has_vbus(s)) {
             ot_usbdev_server_report_connected(s, false);
         }
-        /* @todo Handle FIFO interrupts */
     }
 
+    /* Status interrupts are only active when enabled, we may need to update
+     * them */
+    ot_usbdev_update_status_irqs(s);
     ot_usbdev_update_irqs(s);
 }
 
+/*
+ * Update the value of a CONFIGINx register.
+ *
+ * This function will update the CONFIGINx register after a write and
+ * trigger the necessary changes. If as a result of this change a transfer
+ * completes, the server will be notified.
+ *
+ * @reg Register index
+ * @val32 New value
+ */
+static void ot_usbdev_write_configin(OtUsbdevState *s, hwaddr reg,
+                                     uint32_t val32)
+{
+    uint8_t ep = (uint8_t)(reg - R_CONFIGIN_0);
+    g_assert(ep < USBDEV_PARAM_N_ENDPOINTS);
+
+    /*
+     * @todo clarify what writing 1 to SENDING is supposed to do since this
+     * is a hardware signal and not a status bit, doc is unclear.
+     * In this emulated driver, the SENDING bit is never set by the code.
+     */
+
+    /* clear RW1C fields */
+    uint32_t rw1c = val32 & USBDEV_CONFIGIN_RW1C_MASK;
+    s->regs[reg] &= ~rw1c;
+    /* set RW fields */
+    s->regs[reg] &= ~USBDEV_CONFIGIN_RW1C_MASK;
```

Comment:
Sorry, I think I gave you some slightly wrong code, this should be
```suggestion
    s->regs[reg] &= USBDEV_CONFIGIN_RW1C_MASK;
```
I think? Otherwise it clears out all RW1C fields, when it was intended to clear out all RW fields whilst maintaining the RW1C fields.

Reply:
Indeed ;)